carapace-server/src/main/java/org/carapaceproxy/server/Listeners.java

```diff
@@ -154,17 +169,26 @@ private SslContext bootSslContext(NetworkListenerConfiguration listener, SSLCert
                 LOG.log(Level.INFO, "required sslCiphers " + sslCiphers);
                 ciphers = Arrays.asList(sslCiphers.split(","));
             }
-            return SslContextBuilder
-                            .forServer(keyFactory)
-                            .enableOcsp(listener.isOcsp() && OpenSsl.isOcspSupported())
-                            .trustManager(trustManagerFactory)
-                            .sslProvider(SslProvider.OPENSSL)
-                            .protocols(listener.getSslProtocols())
-                            .ciphers(ciphers).build();
+            SslContext sslContext = SslContextBuilder
+                    .forServer(keyFactory)
+                    .enableOcsp(listener.isOcsp() && OpenSsl.isOcspSupported())
+                    .trustManager(trustManagerFactory)
+                    .sslProvider(SslProvider.OPENSSL)
+                    .protocols(listener.getSslProtocols())
+                    .ciphers(ciphers).build();
+
+            if (listener.isOcsp() && OpenSsl.isOcspSupported()) {
+                String certKey = certificate.getId();
+                parent.getOcspStaplingManager().addCertificateForStapling(certKey, chain);
+                Attribute<Object> attr = sslContext.attributes().attr(AttributeKey.valueOf(OCSP_CERTIFICATE_KEY));
+                attr.set(certKey);
```

Comment:
if we have the Certificate Object in memory you can use it as key and not the static number assigned by the user.
In fact the OCSP Response is only useful for the specific certificate, 
this way we will handle the case of certificate renewal